api/service/legacysync/downloader/client.go

```diff
@@ -57,6 +57,23 @@ func (client *Client) GetBlockHashes(startHash []byte, size uint32, ip, port str
 	return response
 }
 
+// GetBlocksByHeights gets blocks from peers by calling grpc request.
+func (client *Client) GetBlocksByHeights(heights []uint64, ip, port string) *pb.DownloaderResponse {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	request := &pb.DownloaderRequest{
+		Type:    pb.DownloaderRequest_BLOCKBYHEIGHT,
+		Heights: heights,
+	}
+	request.Ip = ip
```

Comment:
Actually ip and port are only used for Register request. We may not need this data from requests.

Reply:
removed